src/decode.rs

```diff
@@ -2970,27 +2968,30 @@ unsafe fn decode_b_inner(
                     N_INTER_INTRA_PRED_MODES as usize - 1,
                 ) as u8;
                 let wedge_ctx = dav1d_wedge_ctx_lut[bs as usize] as c_int;
-                *b.interintra_type_mut() = INTER_INTRA_BLEND
-                    + rav1d_msac_decode_bool_adapt(
-                        &mut ts.msac,
-                        &mut ts.cdf.m.interintra_wedge[wedge_ctx as usize],
-                    ) as u8;
-                if b.interintra_type() == INTER_INTRA_WEDGE {
+                *b.interintra_type_mut() = if rav1d_msac_decode_bool_adapt(
+                    &mut ts.msac,
+                    &mut ts.cdf.m.interintra_wedge[wedge_ctx as usize],
+                ) {
+                    Some(InterIntraType::Wedge)
+                } else {
+                    Some(InterIntraType::Blend)
+                };
```

Comment:
nit:

```suggestion
                *b.interintra_type_mut() = Some(if rav1d_msac_decode_bool_adapt(
                    &mut ts.msac,
                    &mut ts.cdf.m.interintra_wedge[wedge_ctx as usize],
                ) {
                    InterIntraType::Wedge
                } else {
                    InterIntraType::Blend
                });
```

Reply:
yeah? I generally don't like many lines of logic in data constructors, but feel free to commit that if you prefer

Reply:
How about this? 9626bb7

Reply:
Looks good!